CMSgov/design-system
# [WNMGDS-2725] Add optional analytics tracking to `ThirdPartyExternalLink`

## Summary

- Adds a new `thirdPartyExternalLinkSendsAnalytics` config property that controls whether the `ThirdPartyExternalLink` sends analytics events by default
- Adds full analytics tracking capabilities to `ThirdPartyExternalLink`

## How to test

1. Check out [FOR TESTING IN STORYBOOK (adb711)](https://github.com/CMSgov/design-system/pull/3059/commits/adb7113312d8a16943c5c20ca8eabed9e9ee5c03)
2. Turn on analytics in Storybook and try it out
3. Take a look at the new unit tests

## Checklist
